components/automate-cli/cmd/chef-automate/pullAndGenerateConfig.go

```diff
@@ -460,6 +815,51 @@ func getA2ORCSRootCA(config map[string]*dc.AutomateConfig) string {
 	return ""
 }
 
+func getA2fqdn(config map[string]*dc.AutomateConfig) string {
+	for _, ele := range config {
+		if ele.Global != nil && ele.Global.V1 != nil && len(strings.TrimSpace(ele.Global.V1.Fqdn.Value)) > 0 {
+			return ele.Global.V1.Fqdn.Value
+		}
+	}
+	return ""
+}
+
+func getS3Bucket(config map[string]*dc.AutomateConfig) string {
+	for _, ele := range config {
+		if ele.Global.V1.External.Opensearch != nil && ele.Global.V1.External.Opensearch.Backup != nil && ele.Global.V1.External.Opensearch.Backup.S3 != nil && len(strings.TrimSpace(ele.Global.V1.External.Opensearch.Backup.S3.Bucket.Value)) > 0 {
+			return ele.Global.V1.External.Opensearch.Backup.S3.Bucket.Value
+		}
+	}
+	return ""
+}
+
+func getOsRoleArn(config map[string]*dc.AutomateConfig) string {
+	for _, ele := range config {
+		if ele.Global.V1.External.Opensearch != nil && ele.Global.V1.External.Opensearch.Backup != nil && ele.Global.V1.External.Opensearch.Backup.S3 != nil && ele.Global.V1.External.Opensearch.Backup.S3.Settings != nil && len(strings.TrimSpace(ele.Global.V1.External.Opensearch.Backup.S3.Settings.RoleArn.Value)) > 0 {
+			return ele.Global.V1.External.Opensearch.Backup.S3.Settings.RoleArn.Value
+		}
+	}
+	return ""
+}
+
+func getOsAccessKey(config map[string]*dc.AutomateConfig) string {
```

Comment:
Please improve unit test coverage of these functions. Have small unit tests for individual functions, so you can test all the edge cases for different types of input values.

Reply:
Done